src/main/java/io/r2dbc/postgresql/codec/EnumCodec.java

```diff
@@ -129,18 +129,31 @@ public Parameter encodeNull() {
          * @return this {@link Builder}
          */
         public Builder withEnum(String name, Class<? extends Enum<?>> enumClass) {
+            return withEnum(name, enumClass, RegistrationPriority.LAST);
+        }
+
+        /**
+         * Add a Postgres enum type to {@link Enum} mapping.
+         *
+         * @param name      name of the Postgres enum type
+         * @param enumClass the corresponding Java type
+         * @param priority  the codec registration priority
+         * @return this {@link Builder}
+         */
+        public Builder withEnum(String name, Class<? extends Enum<?>> enumClass, RegistrationPriority priority) {
```

Comment:
Let's keep it for the time being more simple. We probably don't need to control priority on a per-codec basis, it's rather whether all enum codecs should come first/last.

That being said, we don't need to expose `RegistrationPriority` and we can keep `withEnum(…)` signatures as-is.

Reply:
The method `withEnum(String, Class<? extends Enum<?>>, RegistrationPriority)` has deleted